### PR TITLE
Add XcodeBuildToolPlugin Support

### PR DESCRIPTION
**Pull request by denil-ct**
## Overview
This PR adds support for `XcodeBuildToolPlugin` support to the SPM plugin. 

## Motivation

The motivation for adding `XcodeBuildToolPlugin` support is to enable the SPM plugin to be added to the Xcode `Run Build Tool Plugins` build phase. This will allow users to generate Swift code from proto files automatically when building their projects. The `XcodeBuildToolPlugin` protocol defines the interface and behavior for plugins that can be invoked by Xcode during the build process.

### CODE REVIEW

**Comment by denil-ct** [comment]
This pull request contains the XcodeBuildToolPlugin part that we agreed to separate from the previous one. @FranzBusch , could you please review it and give your feedback?

**Comment by thomasvl** [comment]
The overview on the PR mention it still having directory support, is that still in or was that removed?  From the other PR, I thought that part was being removed?

**Comment by denil-ct** [comment]
Sorry, my bad. It was indeed removed. Corrected now.

**Comment by thomasvl** [comment]
> Sorry, my bad. It was indeed removed. Corrected now.

No worries, just wanted to double check.

Added reviewers for you also.

**Comment by denil-ct** [comment]
Sure @FranzBusch , will add the changes there as well.

**Comment by thomasvl** [comment]
Looks like the grpc one has some follow up about docs tweaks, should those happen here also?

**Comment by denil-ct** [comment]
Yes, definitely. Best to make the changes here and then move changes there as well.

**Comment by denil-ct** [comment]
@FranzBusch have made some changes to the documentation. Let me know if something is amiss.

**Comment by denil-ct** [comment]
Done @FranzBusch

**Review by FranzBusch — APPROVED** [review]
LGTM! @glbrntt can you check the docs as well since we are going to reflect them to grpc

File: Sources/protoc-gen-swift/Docs.docc/spm-plugin.md [hunk 1]
```diff
@@ -63,9 +51,25 @@ let package = Package(
 
 ### Configuring the plugin
 
-Lastly, after you have added the `.proto` files and modified your `Package.swift` manifest, you can now
-configure the plugin to invoke the `protoc` compiler. This is done by adding a `swift-protobuf-config.json`
-to the root of your target's source folder. An example configuration file looks like this:
+Configuring the plugin is done by adding a `swift-protobuf-config.json` file anywhere in your target's sources. 
+Before you start configuring the plugin, you need to add the `.proto` files to your sources. You should also commit these
+files to your git repository since the generated types are now generated on demand.
+Its also important to note that the proto files in your configuration should be in
```
Comment (glbrntt) [inline Sources/protoc-gen-swift/Docs.docc/spm-plugin.md:57]:
```suggestion
It's also important to note that the proto files in your configuration should be in
```

**Review by glbrntt — APPROVED** [review]
Tiny nit but docs look good otherwise.

File: Sources/protoc-gen-swift/Docs.docc/spm-plugin.md (hunk 1 above)
Comment (denil-ct) [reply Sources/protoc-gen-swift/Docs.docc/spm-plugin.md]: Done, thanks for pointing it out.

**Comment by FranzBusch** [comment]
@thomasvl If you are happy then let's get this merged. @denil-ct Once this is merged, could you open a cherry pick PR to the `1_x_release_branch` branch so that we can get it into the next release and also please update the gRPC PR. Thanks for your contribution!

**Review by thomasvl — APPROVED** [review]

**Comment by thomasvl** [comment]
Thanks again, merged onto `main` now.

**Comment by denil-ct** [comment]
> Once this is merged, could you open a cherry pick PR to the 1_x_release_branch branch so that we can get it into the next release

The release branch is quite a lot of commits behind main. Should I proceed with cherrypicking or are all commits pulled into release branch regardless? @thomasvl @FranzBusch

**Comment by thomasvl** [comment]
main is working towards 2.0, so just cherry pick this one change.

**Comment by denil-ct** [comment]
https://github.com/apple/swift-protobuf/pull/1389 is missing from release branch hence causing conflicts for me. Should I cherrypick that PR as well or should I resolve conflicts for my changes alone?

@FranzBusch @thomasvl

**Comment by thomasvl** [comment]
I believe you'll have to patch things up to work around it, otherwise it would be breaking change on the 1.0 branch. ☹️ 

@FranzBusch Thoughts?

**Pull request merged**